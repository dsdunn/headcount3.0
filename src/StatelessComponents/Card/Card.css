.card {
  height: 300px;
  width: 150px;
  border: 1px solid blue;
}

.district {
  height: 20px;
}

.card ul {
  list-style: none;
  padding-left: 15%;
  padding-right: 15%;
}

.card .year {
  width: 50%;
  text-align: right;
  margin: auto 8px;
}

.card .percent {
  width: 50%;
  text-align: left;
  margin: auto 8px;
}

.district {
  text-align: center;
}

.low {
  background-color: #f4aca8;
}