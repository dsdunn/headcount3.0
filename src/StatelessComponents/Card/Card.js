import React from 'react';

const Card = ({ district }) => {

  const generateList = (stats) => {
    const years = Object.keys(stats);
    years.map(year => {
      return (
        <li>{year}<span className="percent">{stats[year]}</span></li>
        )
    })

  }

  return(
    <div>
      <h5 className="district">Some District</h5>
      <ul>
        {() => generateList(district.stats)}
      </ul>
    </div>
    )
}

export default Card;