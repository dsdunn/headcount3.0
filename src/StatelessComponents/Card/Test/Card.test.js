import React from 'react';
import { shallow } from 'enzyme';
import Card from '../Card.js';

describe('Card', () => {
  let wrapper;
  let mockDistrict;

  it('should match a snapshot', () => {
    mockDistrict = {
      location:'turing', 
      stats:{2007: 0.313}
    }

    wrapper = shallow(<Card district={mockDistrict}
                      unselect={jest.fn()}
                      selectDistrict={jest.fn()}
                      selected={false}
                      />);

    expect(wrapper).toMatchSnapshot();
  });

  it('toggleSelected should call selectDistrict if district.selected is false', () => {
    mockDistrict = {
      location:'turing', 
      stats:{2007: 0.313}
    }
    const mockSelectDistrict = jest.fn();
    wrapper = shallow(<Card 
                      district={mockDistrict}
                      unselect={jest.fn()}
                      selectDistrict={mockSelectDistrict}
                      selected={false}
                      />)

    wrapper.find('.card').simulate('click');

    expect(mockSelectDistrict).toHaveBeenCalled();
  })

  it('toggleSelected should call unselect if district.selected is true', () => {

    const mockUnselect = jest.fn();
    mockDistrict = {
      location:'turing', 
      stats:{2007: 0.313}
    }

    wrapper = shallow(<Card 
                      district={mockDistrict}
                      unselect={mockUnselect}
                      selectDistrict={jest.fn()}
                      selected={true}
                      />)

    wrapper.find('.card').simulate('click');

    expect(mockUnselect).toHaveBeenCalled();
  })

  
})