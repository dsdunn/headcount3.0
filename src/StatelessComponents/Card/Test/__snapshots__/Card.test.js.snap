// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    district={
        Object {
            "location": "turing",
            "stats": Object {
              "2007": 0.313,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5
          className="district"
>
          turing
</h5>,
        <ul>
          <li
                    className="low"
          >
                     
                    <span
                              className="year"
                    >
                              2007
                    </span>
                    <span
                              className="percent"
                    >
                              0.313
                    </span>
          </li>
</ul>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "turing",
          "className": "district",
        },
        "ref": null,
        "rendered": "turing",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="low"
>
               
              <span
                            className="year"
              >
                            2007
              </span>
              <span
                            className="percent"
              >
                            0.313
              </span>
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "2007",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                <span
                  className="year"
>
                  2007
</span>,
                <span
                  className="percent"
>
                  0.313
</span>,
              ],
              "className": "low",
            },
            "ref": null,
            "rendered": Array [
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "2007",
                  "className": "year",
                },
                "ref": null,
                "rendered": "2007",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 0.313,
                  "className": "percent",
                },
                "ref": null,
                "rendered": 0.313,
                "type": "span",
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5
            className="district"
>
            turing
</h5>,
          <ul>
            <li
                        className="low"
            >
                         
                        <span
                                    className="year"
                        >
                                    2007
                        </span>
                        <span
                                    className="percent"
                        >
                                    0.313
                        </span>
            </li>
</ul>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "turing",
            "className": "district",
          },
          "ref": null,
          "rendered": "turing",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="low"
>
                 
                <span
                                className="year"
                >
                                2007
                </span>
                <span
                                className="percent"
                >
                                0.313
                </span>
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "2007",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  <span
                    className="year"
>
                    2007
</span>,
                  <span
                    className="percent"
>
                    0.313
</span>,
                ],
                "className": "low",
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "2007",
                    "className": "year",
                  },
                  "ref": null,
                  "rendered": "2007",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 0.313,
                    "className": "percent",
                  },
                  "ref": null,
                  "rendered": 0.313,
                  "type": "span",
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
